test/testdata/infer/generics/bounds.rb

```diff
@@ -12,56 +12,53 @@ class A1
 
   # should fail: T2 mentions another type member
   T2 = type_member(lower: T1)
-     # ^^^^^^^^^^^^^^^^^^^^^^ error: Only `:fixed` type members are supported
 
   # should fail: T3 mentions another type member
   T3 = type_member(lower: T2, upper: T1)
-     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Only `:fixed` type members are supported
+     # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: `A1::T2` is not a subtype of `A1::T1`
```

Comment:
IRLd that this needs prohibiting with a different error

Reply:
This is prohibited with a different error now.